internal/upload/oci/upload.go

```diff
@@ -0,0 +1,140 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"github.com/oracle/oci-go-sdk/v51/common"
+	"github.com/oracle/oci-go-sdk/v51/core"
+	"github.com/oracle/oci-go-sdk/v51/identity"
+	"github.com/oracle/oci-go-sdk/v51/objectstorage"
+	"github.com/oracle/oci-go-sdk/v51/objectstorage/transfer"
+	"os"
+)
+
+type Uploader interface {
+	Upload(name string, bucketName string, namespace string, file *os.File) error
+	GetBucket(bucketName string, namespace string) (objectstorage.Bucket, error)
+}
+
+type ImageCreator interface {
+	Create(imageName, bucketName, namespace string) (string, error)
+}
+
+type Client struct {
+	Uploader
+	ImageCreator
+	ociClient
+}
+
+// Upload uploads a file into an objectName under the bucketName in the namespace.
+func (c Client) Upload(objectName string, bucketName string, namespace string, file *os.File) error {
+	uploadManager := transfer.NewUploadManager()
+	getBucket, err := c.GetBucket(bucketName, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to find a bucket with name %s under namespace %s: %w", bucketName, namespace, err)
+	}
+	req := transfer.UploadFileRequest{
+		UploadRequest: transfer.UploadRequest{
+			NamespaceName:                       getBucket.Namespace,
+			BucketName:                          getBucket.Name,
+			ObjectName:                          common.String(objectName),
+			PartSize:                            common.Int64(128 * 1024 * 1024),
+			CallBack:                            func(multiPartUploadPart transfer.MultiPartUploadPart) { /* no-op*/ },
+			ObjectStorageClient:                 &c.storageClient,
+			EnableMultipartChecksumVerification: common.Bool(true),
+		},
+		FilePath: file.Name(),
+	}
+
+	ctx := context.Background()
+	resp, err := uploadManager.UploadFile(ctx, req)
+
+	if err != nil && resp.IsResumable() {
```

Comment:
Since I don't think a user can explicitly set that then I'm not sure the error would add something. 
Reading the sdk code I see that if an upload fits into a single upload part (each part is 50Mb) then it is not resumable